src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/create-video-unit/create-video-unit.component.ts

```diff
@@ -0,0 +1,60 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { VideoUnit } from 'app/entities/lecture-unit/videoUnit.model';
+import { VideoUnitFormData } from 'app/lecture/lecture-unit/lecture-unit-management/video-unit-form/video-unit-form.component';
+import { VideoUnitService } from 'app/lecture/lecture-unit/lecture-unit-management/videoUnit.service';
+import { onError } from 'app/shared/util/global.utils';
+import { JhiAlertService } from 'ng-jhipster';
+import { finalize } from 'rxjs/operators';
+
+@Component({
+    selector: 'jhi-create-video-unit',
+    templateUrl: './create-video-unit.component.html',
+    styles: [],
+})
+export class CreateVideoUnitComponent implements OnInit {
+    videoUnitToCreate: VideoUnit = new VideoUnit();
+    isLoading: boolean;
+    lectureId: number;
+    courseId: number;
+
+    constructor(private activatedRoute: ActivatedRoute, private router: Router, private videoUnitService: VideoUnitService, private alertService: JhiAlertService) {}
+
+    ngOnInit(): void {
+        this.activatedRoute.paramMap.subscribe((params) => {
+            this.lectureId = Number(params.get('lectureId'));
+            this.courseId = Number(params.get('courseId'));
+        });
+        this.videoUnitToCreate = new VideoUnit();
+    }
+
+    createAttachmentUnit(formData: VideoUnitFormData) {
```

Comment:
I guess this should be createVideoUnit, right?

Reply:
```suggestion
    createVideoUnit(formData: VideoUnitFormData) {
```

Reply:
done in b2ad6f08755d798fe85237719dc8f1e8de51e93e